From ae78e663adc57c049ccfad919c5349bf351e1775 Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Wed, 19 Jul 2023 11:40:52 +0530
Subject: [PATCH] crypto: testmgr: Disable zero XTS key support

This patch disables zero XTS key test cases. The same
is not supported in ipq9574 and ipq6018 platform and
so this is disabled.

(cherry picked from commit 8a8a81ca018ca427aecbb74a194819060b4c581e)
Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>

Change-Id: I935afa85e4a9b6b62513f08cba4d651d53b61d51
---
 crypto/Kconfig   | 7 -------
 crypto/testmgr.h | 4 ++--
 2 files changed, 2 insertions(+), 9 deletions(-)

diff --git a/crypto/Kconfig b/crypto/Kconfig
index 79dbaef3145a..b99d34d6e7d1 100644
--- a/crypto/Kconfig
+++ b/crypto/Kconfig
@@ -229,13 +229,6 @@ config CRYPTO_TEST
 	help
 	  Quick & dirty crypto test module.
 
-config CRYPTO_NO_AES_CTR_UNEVEN_DATA_LEN_SUPPORT
-	bool "Disable support for uneven data length test cases"
-	default y
-	depends on CRYPTO_TEST
-	help
-	  Disable support for uneven data length cases
-
 config CRYPTO_NO_AES_XTS_ZERO_KEY_SUPPORT
 	bool "Disable support for zero key test cases"
 	default y
diff --git a/crypto/testmgr.h b/crypto/testmgr.h
index f8fba7da99f9..5f6d8d2cd16d 100644
--- a/crypto/testmgr.h
+++ b/crypto/testmgr.h
@@ -18484,9 +18484,9 @@ static const struct cipher_testvec aes_lrw_tv_template[] = {
 static const struct cipher_testvec aes_xts_tv_template[] = {
 	/* http://grouper.ieee.org/groups/1619/email/pdf00086.pdf */
 #ifndef CONFIG_CRYPTO_NO_AES_XTS_ZERO_KEY_SUPPORT
-		/*
+	/*
 	 * zero-key for AES XTS is not supported in ipq6018 and ipq9574
-	 */
+         */
 	{ /* XTS-AES 1 */
 		.key    = "\x00\x00\x00\x00\x00\x00\x00\x00"
 			  "\x00\x00\x00\x00\x00\x00\x00\x00"
-- 
2.34.1

